File: Backend/DTOs/PedidoDto.cs
namespace Pizzaria.DTOs;

public struct PedidoDto
{
    public int ClienteId { get; set; }
    public List<PizzaPedidoDto> Pizzas { get; init; }
    public Dictionary<string, int>? Acompanhamentos { get; init; }
     
    public PedidoDto(int clienteId, List<PizzaPedidoDto> pizzas,  
                    Dictionary<string, int>? acompanhamentos) 
    {
        ClienteId = clienteId;
        Pizzas = pizzas;
        Acompanhamentos = acompanhamentos;
    }
}  

File: Frontend/src/app/models/dtos/PedidoDto.ts
import {PizzaPedidoDto} from "./PizzaPedidoDto";
import {PedidoFinal} from "../entities/PedidoFinal";
import {AcompanhamentoPedidoDto} from "./AcompanhamentoPedidoDto";

export class PedidoDto {
  clienteId: number = 0;
  pizzas : PizzaPedidoDto[] = []
  acompanhamentos : AcompanhamentoPedidoDto[] = [];

  constructor(pedidoFinal: PedidoFinal) {
    pedidoFinal.pizzas.forEach(p => {
      const sabores = p.sabores.map(s =>  s.nome);
      let pizzaPedidoDto = new PizzaPedidoDto(sabores, p.tamanho.nome, p.quantidade);
      this.pizzas.push(pizzaPedidoDto);
    });
    pedidoFinal.acompanhamentos.forEach(a => {
      this.acompanhamentos.push(new AcompanhamentoPedidoDto(a.acompanhamento.nome, a.quantidade));
    });
    this.clienteId = pedidoFinal.cliente.id;
  }
}
